File: Java/Array-1/rotateLeft3.java

public class array1 {
	
	// given an array of ints length 3, return an array with the elements "rotated left" so {1, 2, 3} yields {2, 3, 1}
	public int[] rotateLeft3(int[] nums) {
		
		int[] newArray = new int[3];
		
		newArray[0] = nums[1];
		newArray[1] = nums[2];
		newArray[2] = nums[0];
		
		return newArray;
		
	}
	
	public static void main (String[] args) {
		
		array1 o = new array1();
		
		int[] x = {1, 2, 3};
		int[] y = {5, 11, 9};
		int[] z = {7, 0, 0};
		int[] temp;
		
		temp = o.rotateLeft3(x);
		System.out.println("[" + temp[0] + ", " + temp[1] + ", " + temp[2] + "]");
		
		temp = o.rotateLeft3(y);
		System.out.println("[" + temp[0] + ", " + temp[1] + ", " + temp[2] + "]");
		
		temp = o.rotateLeft3(z);
		System.out.println("[" + temp[0] + ", " + temp[1] + ", " + temp[2] + "]");
		
	}

}

File: JavaScript/Array-1/rotateLeft3.js

// given an array of ints length 3, return an array with the elements 'rotated left' so {1, 2, 3} yields {2, 3, 1}
const rotateLeft3 = (nums) => {
    const array = [];
    
    array.push(nums[1]);
    array.push(nums[2]);
    array.push(nums[0]);
    
    return array;
};

console.log(rotateLeft3([1, 2, 3]));
console.log(rotateLeft3([5, 11, 9]));
console.log(rotateLeft3([7, 0, 0]));
